### ISSUE TITLE
BAZEL_GEN_ROOT in standalone builds isn't substituted on Linux, causing build failures

### ISSUE THREAD

**Issue by mattgodbolt** [bazelbuild/rules_foreign_cc#312]
Due to issues seen in #306 building dependent libraries does not work: the `BAZEL_GEN_ROOT` is not substituted leading to build errors down the line, when using `--spawn_strategy=standalone`. The same build works with sandboxing (but use of sandboxing with remote caching is discouraged).

An example:

--- WORKSPACE
```
load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

all_content = """filegroup(name = "all", srcs = glob(["**"]), visibility = ["//visibility:public"])"""

git_repository(
    name = "rules_foreign_cc",
    commit = "c3d5405cbc570257e7c9f75f902fab42241e6a53",
    remote = "https://github.com/bazelbuild/rules_foreign_cc.git",
    shallow_since = "1565168152 +0200",
)

load("@rules_foreign_cc//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")

rules_foreign_cc_dependencies()

http_archive(
    name = "openssl",
    build_file_content = all_content,
    sha256 = "f56dd7d81ce8d3e395f83285bd700a1098ed5a4cb0a81ce9522e41e6db7e0389",
    strip_prefix = "openssl-OpenSSL_1_1_0h",
    url = "https://github.com/openssl/openssl/archive/OpenSSL_1_1_0h.tar.gz",
)
http_archive(
    name = "curl",
    build_file_content = all_content,
    sha256 = "4376ac72b95572fb6c4fbffefb97c7ea0dd083e1974c0e44cd7e49396f454839",
    strip_prefix = "curl-7.65.3",
    urls = ["https://curl.haxx.se/download/curl-7.65.3.tar.gz"],
)
```

--- BUILD
```
load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")

configure_make(
    name = "openssl",
    configure_command = "config",
    lib_source = "@openssl//:all",
    static_libraries = [
        "libssl.a",
        "libcrypto.a",
    ],
    visibility = ["//visibility:public"],
)

configure_make(
    name = "curl",
    lib_source = "@curl//:all",
    static_libraries = ["libcurl.a"],
    visibility = ["//visibility:public"],
    deps = [
        ":openssl",
    ],
)
```

Run: `$ bazel build //...`

Output:

```
~/bazel-rulescc-bug $ bazel build --spawn_strategy=standalone //...
INFO: Analyzed 2 targets (16 packages loaded, 9964 targets configured).
INFO: Found 2 targets...
INFO: From CcConfigureMakeRule openssl/include:

ERROR: /home/mgodbolt/bazel-rulescc-bug/BUILD:14:1: error executing shell command: '/bin/bash -c function cleanup_function() {
local ecode=$?
if [ $ecode -eq 0 ]; then
cleanup_on_success
...
_=/usr/bin/env
__________________________sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/6a5050bedf5d8d30783a2db385994fa4/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sedCtBrk2: Permission denied
sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/6a5050bedf5d8d30783a2db385994fa4/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sed3p6B20: Permission denied
sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/6a5050bedf5d8d30783a2db385994fa4/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sedDNqnO0: Permission denied
PKG_CONFIG_PATH=:/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/6a5050bedf5d8d30783a2db385994fa4/execroot/__main__/bazel-out/k8-fastbuild/bin/copy_openssl/openssl/lib/pkgconfig
checking whether to enable maintainer-specific portions of Makefiles... no
...
  CC       vtls/libcurl_la-schannel_verify.lo
  CC       vtls/libcurl_la-sectransp.lo
  CC       vtls/libcurl_la-gskit.lo
  CC       vtls/libcurl_la-mbedtls.lo
  CC       vtls/libcurl_la-mesalink.lo
  CCLD     libcurl.la
../libtool: line 7566: cd: BAZEL_GEN_ROOT/openssl/lib: No such file or directory
libtool:   error: cannot determine absolute directory name of 'BAZEL_GEN_ROOT/openssl/lib'
make[2]: *** [Makefile:1184: libcurl.la] Error 1
make[2]: Leaving directory '/tmp/tmp.n3iYKLwYV2/lib'
make[1]: *** [Makefile:1023: all] Error 2
make[1]: Leaving directory '/tmp/tmp.n3iYKLwYV2/lib'
make: *** [Makefile:922: all-recursive] Error 1

_____ END BUILD LOGS _____
rules_foreign_cc: Build script location: bazel-out/k8-fastbuild/bin/curl/logs/Configure_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-fastbuild/bin/curl/logs/Configure.log

INFO: Elapsed time: 227.351s, Critical Path: 226.17s
INFO: 1 process: 1 local.
FAILED: Build did NOT complete successfully
```

Note the `sed` errors early on, which are the cause of the issue (see #306). Errors in `replace_in_path` are dropped, it seems.

**Comment by irengrig** [comment]
Thank you for reporting the problem.
#324 is in review for solving the original problem with sed.

**Comment by irengrig** [comment]
Fixed by #324
